pandas/io/formats/format.py

```diff
@@ -1216,10 +1222,15 @@ def _format_strings(self) -> List[str]:
         else:
             float_format = self.float_format
 
+        quote_strings = self.quoting is not None and self.quoting != QUOTE_NONE
         formatter = (
             self.formatter
             if self.formatter is not None
-            else (lambda x: pprint_thing(x, escape_chars=("\t", "\r", "\n")))
+            else (
+                lambda x: pprint_thing(
+                    x, escape_chars=("\t", "\r", "\n"), quote_strings=quote_strings
+                )
+            )
```

Comment:
as the ternary expression is getting more complex, an if else may now be more readable. maybe use partial instead of lambda and maybe move quote_strings assignment inside the relevant if else block.